homeassistant/components/homekit_controller/climate.py

```diff
@@ -56,15 +89,219 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
+            CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
+        ):
+            return self.service[
+                CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
+            ].minStep
+        if state == TargetHeaterCoolerStateValues.HEAT and self.service.has(
+            CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
+        ):
+            return self.service[
+                CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
+            ].minStep
+        return None
+
+    @property
+    def min_temp(self):
+        """Return the minimum target temp."""
+        state = self.service.value(CharacteristicsTypes.TARGET_HEATER_COOLER_STATE)
+        if state == TargetHeaterCoolerStateValues.COOL and self.service.has(
+            CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
+        ):
+            return self.service[
+                CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
+            ].minValue
+        if state == TargetHeaterCoolerStateValues.HEAT and self.service.has(
+            CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
+        ):
+            return self.service[
+                CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
+            ].minValue
+        return super().min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum target temp."""
+        state = self.service.value(CharacteristicsTypes.TARGET_HEATER_COOLER_STATE)
+        if state == TargetHeaterCoolerStateValues.COOL and self.service.has(
+            CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
+        ):
+            return self.service[
+                CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD
+            ].maxValue
+        if state == TargetHeaterCoolerStateValues.HEAT and self.service.has(
+            CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
+        ):
+            return self.service[
+                CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD
+            ].maxValue
+        return super().max_temp
+
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation."""
+        # This characteristic describes the current mode of a device,
+        # e.g. a thermostat is "heating" a room to 75 degrees Fahrenheit.
+        # Can be 0 - 3 (Off, Idle, Heat, Cool)
+        value = self.service.value(CharacteristicsTypes.CURRENT_HEATER_COOLER_STATE)
+        return CURRENT_HEATER_COOLER_STATE_HOMEKIT_TO_HASS.get(value)
+
+    @property
+    def hvac_mode(self):
+        """Return hvac operation ie. heat, cool mode."""
+        # This characteristic describes the target mode
+        # E.g. should the device start heating a room if the temperature
+        # falls below the target temperature.
+        # CCan be 0 - 2 (Auto, Heat, Cool)
+        if (
+            self.service.has(CharacteristicsTypes.ACTIVE)
+            and self.service.value(CharacteristicsTypes.ACTIVE)
```

Comment:
`self.service.value` returns None if the characterstic is not present (or you can provide a default value) - so you can get rid of this `has` check here.